components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/internal/HttpClientProvider.java

```diff
@@ -19,17 +19,20 @@
 import org.sonatype.nexus.httpclient.HttpClientManager;
 
 import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.eclipse.sisu.Typed;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
- * {@link HttpClient} provider via {@link HttpClientManager}.
+ * {@link HttpClient} (and {@link CloseableHttpClient}) provider via {@link HttpClientManager}.
  *
  * @since 3.0
  */
 @Named
+@Typed({HttpClient.class, CloseableHttpClient.class})
 public class HttpClientProvider
-    implements Provider<HttpClient>
+    implements Provider<CloseableHttpClient>
```

Comment:
Just for information: this will still work with `Provider<HttpClient>` as the `@Typed` details take precedent, but using the more descriptive type in the Provider API is better as you also get the runtime checks.